builtin/logical/mongodb/util.go

```diff
@@ -0,0 +1,82 @@
+package mongodb
+
+import (
+	"crypto/tls"
+	"errors"
+	"net"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"gopkg.in/mgo.v2"
+)
+
+// Unfortunately, mgo doesn't support the ssl parameter in its MongoDB URI parsing logic, so we have to handle that
+// ourselves. See https://github.com/go-mgo/mgo/issues/84
+func parseMongoURI(rawUri string) (*mgo.DialInfo, error) {
+	uri, err := url.Parse(rawUri)
+	if err != nil {
+		return nil, err
+	}
+
+	info := mgo.DialInfo{
+		Addrs:    strings.Split(uri.Host, ","),
```

Comment:
Just to check -- have you tested that `url.Parse` handles sets of hosts correctly? So if you have multiple hosts split with commans, `uri.Host` contains what you want?